index.js

```diff
@@ -263,25 +269,37 @@ ZongJi.prototype.start = function (options = {}) {
   Promise.all(promises)
     .then(() => {
       this.BinlogClass = initBinlogClass(this);
-      this.ready = true;
-      this.emit('ready');
-
-      this.connection._protocol._enqueue(new this.BinlogClass(binlogHandler));
+      if (!this.stopped) {
+        this.connection._protocol._enqueue(new this.BinlogClass(binlogHandler));
+        this.ready = true;
+        this.emit('ready');
+      }
     })
     .catch((err) => {
       this.emit('error', err);
     });
 };
 
 ZongJi.prototype.stop = function () {
-  // Binary log connection does not end with destroy()
-  this.connection.destroy();
-  this.ctrlConnection.query('KILL ' + this.connection.threadId, () => {
-    if (this.ctrlConnectionOwner) {
-      this.ctrlConnection.destroy();
-    }
-    this.emit('stopped');
-  });
+  if (!this.stopped) {
+    this.stopped = true;
+    // Binary log connection does not end with destroy()
+    this.connection.destroy();
+    this.ctrlConnection.query('KILL ' + this.connection.threadId, (err, result) => {
+      if (this.ctrlConnectionOwner) {
+        this.ctrlConnection.destroy();
+      }
+      this.emit('stopped');
+    });
+  }
+};
+
+ZongJi.prototype.pause = function () {
+  this.connection.pause();
```

Comment:
Should these methods check if the ZonJi instance has been stopped before running?